File: src/exemplos/05_EstruturasDeDados/04_Vetor/12-main.c
/**      @file: 12-main.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo de má utilização de ponteiros. */

#include <stdio.h>

int main(int argc, char* argv[]) {
    /* A ideia é cumprimentar o usuário... */
    printf("Boa tarde, %s.\n", argv[1]);

    return 0;
}

/* É perigoso acessar memória além da planejada... Tente:
 *   printf("%s\n", argv[2]);
 *   printf("%s\n", argv[3]);
 *   printf("%s\n", argv[5]);
 *   printf("%s\n", argv[50]);
 *   printf("%s\n", argv[5000]); */

File: src/exemplos/05_EstruturasDeDados/04_Vetor/12-main.py
#  -*- coding: utf-8 -*-
##   @package: 12-main.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplo de má utilização de ponteiros.


import sys


# A ideia é cumprimentar o usuário...
print('Boa tarde, %s.' % sys.argv[1])


# É perigoso acessar memória além da planejada... Por isso Python não deixa o
# usuário acessar índices incorretos.
# print 'Boa tarde, %s.' % sys.argv[2]
# print 'Boa tarde, %s.' % sys.argv[3]
# print 'Boa tarde, %s.' % sys.argv[5]
# print 'Boa tarde, %s.' % sys.argv[50]
# print 'Boa tarde, %s.' % sys.argv[5000]
